pyccl/baryons.py

```diff
@@ -0,0 +1,137 @@
+            return new
+
+
+class BaryonicEffectsBCM(BaryonicEffects):
+    """The BCM model boost factor for baryons.
+
+    .. note:: BCM stands for the "baryonic correction model" of Schneider &
+              Teyssier (2015; https://arxiv.org/abs/1510.06034). See the
+              `DESC Note <https://github.com/LSSTDESC/CCL/blob/master/doc\
+/0000-ccl_note/main.pdf>`_
+              for details.
+
+    .. note:: The boost factor is applied multiplicatively so that
+              :math:`P_{\\rm corrected}(k, a) = P(k, a)\\, f_{\\rm bcm}(k, a)`.
+
+    Args:
+        log10Mc (:obj:`float`): logarithmic mass scale of hot
+            gas suppression. Defaults to 14.08.
+        eta_b (:obj:`float`): ratio of escape to ejection radii (see
+            Teyssier et al. 2015). Defaults to 0.5.
+        k_s (:obj:`float`): Characteristic scale (wavenumber) of
+            the stellar component. Defaults to 55.0.
+    """
+    __repr_attrs__ = ("log10Mc", "eta_b", "k_s")
+
+    def __init__(self, log10Mc=14.08, eta_b=0.5, k_s=55.0):
+        self.log10Mc = log10Mc
+        self.eta_b = eta_b
+        self.k_s = k_s
+        self.apply_baryons = self._apply_boost
+
+    def boost_factor(self, cosmo, k, a):
+        """The BCM model boost factor for baryons.
+
+        Args:
+            cosmo (:class:`~pyccl.core.Cosmology`): Cosmological parameters.
+            k (float or array_like): Wavenumber; Mpc^-1.
+            a (float): Scale factor.
+
+        Returns:
+            float or array_like: Correction factor to apply to
+                the power spectrum.
+        """
+        k_use = np.atleast_1d(k)
+
+        z = 1./a - 1.
+        kh = k_use / cosmo['h']
+        b0 = 0.105*self.log10Mc - 1.27
+        bfunc = b0 / (1. + (z/2.3)**2.5)
+        bfunc4 = (1-bfunc)**4
+        kg = 0.7 * bfunc4 * self.eta_b**(-1.6)
+        gf = bfunc / (1 + (kh/kg)**3) + 1. - bfunc
+        scomp = 1 + (kh / self.k_s)**2
+        fka = gf * scomp
+
+        if np.ndim(k) == 0:
+            fka = fka[0]
+        return fka
+
+    def update_parameters(self, log10Mc=None, eta_b=None, k_s=None):
+        """Update BCM parameters.
+
+        Args:
+            log10Mc (:obj:`float`): logarithmic mass scale of hot
+                gas suppression. Defaults to 14.08.
+            eta_b (:obj:`float`): ratio of escape to ejection radii (see
+                Teyssier et al. 2015). Defaults to 0.5.
+            k_s (:obj:`float`): Characteristic scale (wavenumber) of
+                the stellar component. Defaults to 55.0.
+        """
+        if log10Mc is not None:
+            self.log10Mc = log10Mc
+        if eta_b is not None:
+            self.eta_b = eta_b
+        if k_s is not None:
+            self.k_s = k_s
+
+    @unlock_instance(mutate=True, argv=2)
+    def _apply_boost(self, cosmo, pk2d, in_place=False):
+        # Applies boost factor
+        if not isinstance(pk2d, Pk2D):
+            raise TypeError("pk2d must be a Pk2D object")
```

Comment:
No need to type-check this, it will fail with a useful error in the next line. There's a bunch of needless type-checks all over CCL which I am also planning to remove. If you really need to use type-checking, simply add a type hint in the arguments, so that users can run it through `mypy`.